ambry-named-mysql/src/main/java/com/github/ambry/named/MySqlNamedBlobDb.java

```diff
@@ -113,16 +113,17 @@ class MySqlNamedBlobDb implements NamedBlobDb {
    */
   // @formatter:off
   private static final String LIST_QUERY_V2 = String.format(""
-          + "SELECT t1.blob_name, t1.blob_id, t1.version, t1.deleted_ts, t1.blob_size, t1.modified_ts "
-          + "FROM named_blobs_v2 t1 "
-          + "INNER JOIN "
-          + "(SELECT account_id, container_id, blob_name, max(version) as version "
-          + "FROM named_blobs_v2 "
-          + "WHERE (account_id, container_id) = (?, ?) AND %1$s "
-          + "  AND (deleted_ts IS NULL OR deleted_ts>%2$S) "
-          + "        GROUP BY account_id, container_id, blob_name) t2 "
-          + "ON (t1.account_id,t1.container_id,t1.blob_name,t1.version) = (t2.account_id,t2.container_id,t2.blob_name,t2.version) "
-          + "WHERE t1.blob_name LIKE ? AND t1.blob_name >= ? ORDER BY t1.blob_name ASC LIMIT ?",STATE_MATCH, CURRENT_TIME);
+      + "SELECT blob_name, blob_id, MAX(version) AS version, deleted_ts, blob_size, modified_ts"
+      + "  FROM named_blobs_v2"
+      + "  WHERE account_id = ?"
+      + "    AND container_id = ?"
+      + "    AND blob_state = %1$s"
+      + "    AND (deleted_ts IS NULL OR deleted_ts > %2$S)"
```

Comment:
If we have two versions, ver1 delete_ts is null, ver2 delete_ts is not null, we will return ver1. Which is not the correct case. We need to return nothing.

Reply:
Good pt. I addressed it using the WITH-clause.

Reply:
`testListNamedBlobs` already checks for it

Reply:
added another test - a better one